clar2wasm/tests/wasm-generation/principal.rs

```diff
@@ -0,0 +1,254 @@
+                 // will return an Ok value.
+                0x1A | 0x15 => Value::okay(expected_principal),
+                0x00..=0x1F => Ok(create_error_construct(0, Some(expected_principal))),
+                _ => Ok(create_error_construct(1, None))
+            }.unwrap();
+
+            let snippet = match contract {
+                Some(ctc) => &format!("(principal-construct? 0x{:02X} {hash_bytes} \"{}\")", version_byte, ctc),
+                None => &format!("(principal-construct? 0x{:02X} {hash_bytes})", version_byte)
+            };
+
+            crosscheck(
+                snippet,
+                Ok(Some(expected)),
+            );
+        }
+    }
+
+    proptest! {
+        #![proptest_config(runtime_config())]
+
+        #[test]
+        fn crosscheck_principal_destruct(
+            version_byte in strategies_for_version_byte(),
+            hash_bytes in buffer(20),
+            contract in "([a-zA-Z](([a-zA-Z0-9]|[-])){0, 30})".prop_flat_map(|name| {
+                prop_oneof![Just(Some(name)), Just(None)]
+            })
+        ) {
+
+            let expected_principal = create_principal(
+                version_byte as u8,
+                &hash_bytes.clone().expect_buff(20).unwrap(),
+                contract.as_deref()
+            );
+
+            let data = contract.map(|ctc| Box::new(
+                Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+                    data: ctc.into_bytes()
+                })))
+            ));
+
+            let expected = match version_byte {
+                // Since tests runs on a Testnet version,
+                // version_bytes single_sig (0x1A) || multi_sig (0x15), for Testnet,
+                // will return an Ok value.
+                0x1A | 0x15 => Value::okay(
+                    TupleData::from_data(vec![
+                        ("hash-bytes".into(), hash_bytes),
+                        ("name".into(), Value::Optional(OptionalData { data })),
+                        (
+                            "version".into(),
+                            Value::Sequence(SequenceData::Buffer(BuffData {
+                                data: vec![version_byte as u8],
+                            })),
+                        ),
+                    ])
+                    .unwrap()
+                    .into()
+                ),
+                0x00..=0x1F => Ok(create_error_destruct(hash_bytes, version_byte as u8, data)),
+                _ => Ok(create_error_destruct(hash_bytes, version_byte as u8, data)),
+            }.unwrap();
+
+            crosscheck(
+                &format!("(principal-destruct? {})", PropValue::from(expected_principal.clone())),
+                Ok(Some(expected)),
+            );
+        }
+    }
+
+    proptest! {
+        #![proptest_config(runtime_config())]
+
+        #[test]
+        fn crosscheck_is_standard(
+            principal in standard_principal().prop_map(PropValue::from)
+        ) {
```

Comment:
With the last refactor we are no longer testing contract principals here.

Reply:
Added `qualified` principal.